src/compat.h

```diff
@@ -676,4 +676,90 @@ pg_sub_s64_overflow(int64 a, int64 b, int64 *result)
 }
 #endif
 
+/* Backport of list_qsort() */
+#if PG11_LT
+typedef int (*list_qsort_comparator)(const void *a, const void *b);
+/*
+ * Sort a list as though by qsort.
+ *
+ * A new list is built and returned.  Like list_copy, this doesn't make
+ * fresh copies of any pointed-to data.
+ *
+ * The comparator function receives arguments of type ListCell **.
+ */
+
+static List *
+new_list(NodeTag type)
+{
+	List *new_list;
+	ListCell *new_head;
+
+	new_head = (ListCell *) palloc(sizeof(*new_head));
+	new_head->next = NULL;
+	/* new_head->data is left undefined! */
```

Comment:
Suggest to use `palloc0` and define `data` field as well. Does not really make a difference performance-wise, and will simplify debugging significantly.

Reply:
This is copied/backported verbatim from Postgres code and only applies to older PG versions (newer versions use PG function directly). I don't think its a good idea to have different code/logic between PG versions.